src/publisher/fragment_logic.py

```diff
@@ -181,24 +187,39 @@ def merge_if_valid(av, quiet=True):
     if invalid and quiet=False, a ValidationError will be raised"""
     return valid(merge_and_preprocess(av), quiet=quiet)
 
-def set_article_json(av, quiet):
+def hash_ajson(merge_result):
+    string = utils.json_dumps(merge_result, indent=None)
+    return hashlib.md5(string.encode('utf-8')).hexdigest()
+
+class Identical(RuntimeError):
+    def __init__(self, msg, av, hashval):
+        super(Identical, self).__init__(msg)
+        self.av = av
+        self.hashval = hashval
+
+# TODO: rename `quiet` to `valid_check` or similar.
+def set_article_json(av, quiet, hash_check=True):
     """updates the article with the result of the merge operation.
     if the result of the merge was valid, the merged result will be saved.
     if invalid, the ArticleVersion instance's article-json will be unset.
     if invalid and quiet=False, a ValidationError will be raised"""
-    log_context = {'article-version': av, 'quiet': quiet}
+    log_context = {'article-version': av, 'quiet': quiet, 'hash_check': hash_check}
```

Comment:
I would check that the changes to this `fragment_logic` module work well with the POST fragment API https://github.com/elifesciences/lax/blob/fa2b4a55a7475dafec83b17d88e48bb9b5e3fe87/src/publisher/api_v2_views.py#L133-L167
For example, there is the additional `Identical` exception to deal with and possibly more. It's quite a similar flow however, an event should be sent if the POST of the fragment updates the article, but not if the hash doesn't change, etc.

Reply:
good catch, I'll investigate

Reply:
fixed up. actually found a case that wasn't handled and fixed that up too.